### PR TITLE
add jpa repository + liquibase support + safe delete impl + custom up…

### PR DESCRIPTION
**Pull request by slaybrute**
…date method + fixed pom.xml

### CODE REVIEW

File: src/main/java/com/example/onlinebookstore/repository/BookRepository.java [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
-
-    List<Book> getAll();
-
-    Optional<Book> findById(Long id);
+public interface BookRepository extends JpaRepository<Book, Long> {
+    @Modifying
+    @Transactional
+    @Query("UPDATE Book b SET b.author = :author, b.title = :title, "
+            + "b.price = :price, b.description = :description, b.coverImage = :coverImage "
+            + "WHERE b.id = :id")
```
Comment (okuzan) [inline src/main/java/com/example/onlinebookstore/repository/BookRepository.java]: to update you can just call .save() on existing entity (with id), no need for this method. You can use mapper to create entity from DTO

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/onlinebookstore/repository/BookRepository.java (hunk 1 above)
Comment (slaybrute) [reply src/main/java/com/example/onlinebookstore/repository/BookRepository.java]: > to update you can just call .save() on existing entity (with id), no need for this method. You can use mapper to create entity from DTO

but it's good choice as well

Comment (slaybrute) [reply src/main/java/com/example/onlinebookstore/repository/BookRepository.java]: I thought It's more explicit approach. Is it ?

**Review by Ivan95kos — APPROVED** [review]
Nice

**Pull request merged**